source/agora/node/Ledger.d

```diff
@@ -222,6 +236,11 @@ public class Ledger
         if (!this.storage.saveBlock(block))
             assert(0);
 
+        foreach (enrollment; block.header.enrollments)
```

Comment:
There is another condition here which @Geod24 mentioned (he mentioned it in real life).

If we have received a Block from the network we try to update the heights of the enrollments that are in the block.

But what if the enrollments are not in the EnrollmentManager? For example, if we received a Block before we received Enrollments via `registerEnrollment. Then `updateEnrolledHeight` will fail, and nodes will have an inconsistent state.

I'm not sure what the best approach is here. Add the enrollment from the block to the enrollment manager via `addEnrollment`? @Geod24

Reply:
>If we received a Block before we received our received Enrollments via `registerEnrollment` 

Can you tell me which part of the time you're talking about?

It creates a block until it receives enrollments at the time it is created.
`makeNewBlock` in `onTXSetExternalized`

However, if there isn't received `enrolment data` until the block is created, the node will create another block.